spec/syntax.md

```diff
@@ -495,6 +497,8 @@ Inside _patterns_,
 whitespace is part of the translatable content and is recorded and stored verbatim.
 Whitespace is not significant outside translatable text, except where required by the syntax.
 
+***NOTE:*** Whitespace **_is_** significant in the `reserved` production and implementations need to be careful not to trim trailing whitespace from reserved sequences.
```

Comment:
As an alternative solution, we could also tie our future hands a little, and explicitly make the trailing whitespace of `reserved` not significant. Realistically, I don't think that would cause any problems, especially as we're already considering trailing whitespace non-significant in other expressions.

Reply:
We can do that, but we would need to change the ABNF. The problem is that all of our other productions are bounded in some way and the exterior whitespace is applied by the optional production `[s]`. Because reserved sequences are not delimited, there is no way to know where the whitespace belongs to the reserved or to the `[s]`. Future standardization would most likely use productions like `name` or `nmtoken` and the resulting trailing whitespace would "emerge" from the murky fog of `reserved` to be captured by `[s]`. Note that this is one of the reasons I originally proposed reserving the sigils to the function-like syntax.

I tend to agree with wanting to make the exterior whitespace not significant, in part because it interferes with user expectations. These two sequences are logically identical:

```
{      $foo :bar    }
{$foo :bar}
```

But these two are not:

```
{    $foo ~bar    }
{$foo ~bar}
```

I think we should fix this. Probably:

```abnf
reserved = reserved-start reserved-body reserved-end
reserved-end = ; reserved-body less SP / HTAB/ CR / LF
```

Reply:
Why do we need to distinguish the trailing spaces of `reserved` in the ABNF? Would it not be enough to include some statement like this in the text on reserved about future use?
> Future uses of _reserved_ expressions MUST NOT assign any significance to trailing whitespace.

Reply:
We should make the parser return the right tokens. See what you think of the revised abnf.